lua/luasnip/nodes/choiceNode.lua

```diff
@@ -45,7 +49,47 @@ function ChoiceNode:init_nodes()
 	self.active_choice = self.choices[1]
 end
 
-local function C(pos, choices, opts)
+---@class LuaSnip.Opts.ChoiceNode: LuaSnip.Opts.Node
+---@field restore_cursor boolean? If set, the currently active node is looked up
+---in the switched-to choice, and the cursor restored to  preserve the current
+---position relative to that node. The node may be found if a `restoreNode` is
+---present in both choice.
+---`false` by default, as enabling might lead to decreased performance.
+---
+---It's possible to override the default by wrapping the `choiceNode`
+---constructor in another function that sets `opts.restore_cursor` to `true` and
+---then using that to construct `choiceNode`s:
+---```lua
+---local function restore_cursor_choice(pos, choices, opts)
+---    if opts then
+---        opts.restore_cursor = true
+---    else
+---        opts = {restore_cursor = true}
+---    end
+---    return c(pos, choices, opts)
+---end
+---```
+---Consider passing this override into `snip_env`.
+---@field node_callbacks table<"change_choice"|"enter"|"leave", fun(node:LuaSnip.Node)>
+---Specify functions to call after changing the choice, or entering or leaving
+---the node. The callback receives the `node` the callback was called on.
+
+---Create a new choiceNode.
+---@param pos integer Jump-index of the node (See
+---[Basics-Jump-Index](../../../DOC.md#jump-index)).
+---@param choices table<integer, LuaSnip.Node|LuaSnip.Node[]> A list of nodes
+---that can be switched between. If a list of nodes is passed as a choice, it
+---will be turned into a snippetNode.
+---Jumpable nodes that generally need a jump-index don't need one inside the
+---choiceNode, they inherit its jump-index. Additionally, one should make sure
+---the cursor has a position to stop at inside every choice, since changing the
```

Comment:
random question: Is that still true? I know I can have a choice be a simple text node and the choice selection will stop on it already (no need for an i(1))

Reply:
Good point, that's an exception, I'll amend the doc :+1:

Reply:
ea7ba6d